packages/amplify-category-auth/index.js

```diff
@@ -109,38 +112,30 @@ async function externalAuthEnable(context, externalCategory, resourceName, requi
         { resourceName: `cognito${sharedId}` },
       ); //eslint-disable-line
   /* eslint-enable */
-  const roles = await context.amplify.executeProviderUtils(
-    context,
-    'awscloudformation',
-    'staticRoles',
-  );
+  const { roles } = defaults;
   const authProps = {
     ...authPropsValues,
     ...roles,
   };
 
   try {
-    const copyJobs = [
-      {
-        dir: pluginDir,
-        template: `provider-utils/awscloudformation/cloudformation-templates/${cfnFilename}`,
-        target: `${targetDir}/${category}/${authProps.resourceName}/${
-          authProps.resourceName
-        }-cloudformation-template.yml`,
-        paramsFile: `${targetDir}/${category}/${authProps.resourceName}/parameters.json`,
-      },
-    ];
-
-    // copy over the files
-    await context.amplify.copyBatch(context, copyJobs, authProps, true, true);
-    if (!context.updatingAuth) {
+    await copyCfnTemplate(context, category, authProps, cfnFilename);
+    saveResourceParameters(
+      context,
+      provider,
+      category,
+      authProps.resourceName,
+      authProps,
+      ENV_SPECIFIC_PARAMS,
+    );
+    if (!authExists) {
       const options = {
         service: 'Cognito',
         providerPlugin: 'awscloudformation',
       };
       await amplify.updateamplifyMetaAfterResourceAdd(category, authProps.resourceName, options);
     }
-    const action = context.updatingAuth ? 'updated' : 'added';
```

Comment:
this updating auth is defined somewhere I believe? Did you check dependency plugins setting it?

Reply:
Its not set anywhere other then the update command for auth. Also auth should be updated if its already present or should be added if its not.